synapse/config/server.py

```diff
@@ -754,501 +751,18 @@ def generate_config_section(
                 compress: false"""
 
             if listeners:
```

Comment:
All of the logic around handling `listeners` looks fairly complicated, given that this is meant to just dump out some default config. Is there anywhere that we actually pass in an explicit list of listeners? If not, we might consider removing this argument from `synapse.config._base.RootConfig.generate_config` altogether.

Reply:
There are some tests that have explicit listeners passed in, but I didn't see any scripts/functions. I suppose it could be removed but I feel nervous about it-I will defer to your expertise.